src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Windows.cs

```diff
@@ -414,16 +400,14 @@ unsafe private static void AsyncWaitForConnectionCallback(uint errorCode, uint n
             asyncResult._isComplete = true;
 
             // The OS does not signal this event.  We must do it ourselves.
-            ManualResetEvent wh = asyncResult._waitHandle;
-            if (wh != null)
+            ManualResetEventSlim wh = asyncResult.WaitHandle;
+            Debug.Assert(!wh.IsSet, "ManualResetEventSlim already signaled!");
+            wh.Set();
+            bool isSet = wh.IsSet;
```

Comment:
This check and the subsequent assert and throw is unnecessary.  (And even if it could fail, it wouldn't result in a Win32 error.)